job_creator.php

```diff
@@ -346,7 +346,8 @@ private function buildDynamicMatrix(
         if ($run['endtoend'] && file_exists('behat.yml')) {
             $job = $this->createJob(0, [
                 'endtoend' => true,
-                'endtoend_suite' => 'root'
+                'endtoend_suite' => 'root',
+                'composer_require_extra' => $simpleMatrix ? '' : 'silverstripe/graphql:^3'
```

Comment:
We only want this where the major version for installer is `4`

Reply:
We don't have any support elsewhere for installer 5 yet

Reply:
There have been bits throughout where we've been careful to have some cross-major support, but if you want to do that separately for whatever reason then I guess that's fine

Reply:
Updated